### PR TITLE
docs: add sql/listinstances example

### PR DESCRIPTION
**Pull request by broady**
@adg

### CODE REVIEW

File: docs/sql/listinstances/listinstances.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package listinstances
```
Comment (adg) [inline docs/sql/listinstances/listinstances.go]: Need copyright

File: docs/sql/listinstances/listinstances.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	if err != nil {
+		return nil, err
+	}
+	// Create the Google Cloud SQL service.
+	service, err := sqladmin.New(client)
+	if err != nil {
+		return nil, err
+	}
+	// Create the request to list instances for the project ID.
+	request := service.Instances.List(projectId)
```
Comment (adg) [inline docs/sql/listinstances/listinstances.go]: make this a one-liner with the next statement?

File: docs/sql/listinstances/listinstances.go (hunk 1 above)
Comment (broady) [inline docs/sql/listinstances/listinstances.go]: Done (and for _test)

File: docs/sql/listinstances/listinstances.go (hunk 2 above)
Comment (broady) [inline docs/sql/listinstances/listinstances.go]: Done. And cleaned up a bunch of other stuff (like using the scope constant, variable names)

**Pull request merged**